tgui/packages/tgui/interfaces/CharacterPreferences/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { useBackend, useLocalState, useSharedState } from '../../backend';
+import {
+  Box,
+  Button,
+  ByondUi,
+  Divider,
+  Flex,
+  Icon,
+  LabeledList,
+  NoticeBox,
+  Section,
+  Stack,
+  Tabs,
+} from '../../components';
+import { Window } from '../../layouts';
+import { CharacterPreferencesData, CharacterPreferencesProfile, CharacterPreferencesTabKeys } from './type';
+import { SavesTab } from './SavesTab';
+import { CharacterTab } from './CharacterTab';
+import { GeneralTab } from './GeneralTab';
+import { GameSettingsTab } from './GameSettingsTab';
+import { Fragment } from 'inferno';
+
+export const CharacterPreferences = (_props, context) => {
+  const { act, data } = useBackend<CharacterPreferencesData>(context);
+  const [menu, setMenu] = useSharedState(context, 'menu', CharacterPreferencesTabKeys.General);
+
+  return (
+    <Window width={600} height={750} title="Character Setup">
+      <Window.Content scrollable>
+        <SavesAndProfile />
+
+        <Tabs>
+          <Tabs.Tab
+            selected={menu === CharacterPreferencesTabKeys.General}
+            onClick={() => setMenu(CharacterPreferencesTabKeys.General)}>
+            General
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={menu === CharacterPreferencesTabKeys.Character}
+            onClick={() => setMenu(CharacterPreferencesTabKeys.Character)}>
+            Appearance
+          </Tabs.Tab>
+          <Tabs.Tab onClick={() => act('occupation-window')}>Occupation</Tabs.Tab>
+          <Tabs.Tab onClick={() => act('traits-window')}>Traits</Tabs.Tab>
+          <Tabs.Tab
+            selected={menu === CharacterPreferencesTabKeys.GameSettings}
+            onClick={() => setMenu(CharacterPreferencesTabKeys.GameSettings)}>
+            Game Settings
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={menu === CharacterPreferencesTabKeys.Saves}
+            onClick={() => setMenu(CharacterPreferencesTabKeys.Saves)}>
+            Cloud Saves
+          </Tabs.Tab>
+        </Tabs>
+        {menu === CharacterPreferencesTabKeys.General || menu === CharacterPreferencesTabKeys.Character ? (
+          <Flex>
+            <Flex.Item grow="1">
+              <Box>
+                {menu === CharacterPreferencesTabKeys.General && <GeneralTab />}
+                {menu === CharacterPreferencesTabKeys.Character && <CharacterTab />}
+              </Box>
+            </Flex.Item>
+            <Flex.Item ml="5px">
```

Comment:
That you have `ml` here implies to me that this should probably be a `Stack` rather than a `Flex`.